tests/blockchain/eth/contracts/main/mining_adjudicator/test_mining_adjudicator.py

```diff
@@ -109,45 +158,46 @@ def test_challenge_cfrag(testerchain, escrow, challenge_contract):
             miner_umbral_public_key_bytes,
             signed_miner_umbral_public_key,
             some_data)
-    tx = challenge_contract.functions.challengeCFrag(*args).transact()
+    tx = adjudicator_contract.functions.evaluateCFrag(*args).transact()
     testerchain.wait_for_receipt(tx)
     # Hash of the data is saved and miner was not slashed
-    assert challenge_contract.functions.challengedCFrags(data_hash).call()
-    assert 1000 == escrow.functions.minerInfo(miner).call()
+    assert adjudicator_contract.functions.evaluatedCFrags(data_hash).call()
+    assert 1000 == escrow.functions.minerInfo(miner).call()[0]
 
-    # Can't challenge miner with data that already was checked
+    # Can't evaluate miner with data that already was checked
     with pytest.raises((TransactionFailed, ValueError)):
-        tx = challenge_contract.functions.challengeCFrag(*args).transact()
+        tx = adjudicator_contract.functions.evaluateCFrag(*args).transact()
         testerchain.wait_for_receipt(tx)
 
-    # Challenge using bad data
-    metadata = os.urandom(34)
-    capsule, cfrag = fragments(metadata)
-    capsule_bytes = capsule.to_bytes()
-    cfrag_bytes = cfrag.to_bytes()
-    hash_ctx = hashes.Hash(hashes.SHA256(), backend=backend)
-    hash_ctx.update(capsule_bytes + cfrag_bytes)
-    data_hash = hash_ctx.finalize()
-    capsule_signature_by_requester = sign_data(capsule_bytes, requester_umbral_private_key)
-    capsule_signature_by_requester_and_miner = sign_data(capsule_signature_by_requester, miner_umbral_private_key)
-    cfrag_signature_by_miner = sign_data(cfrag_bytes, miner_umbral_private_key)
-    args = (capsule_bytes,
-            capsule_signature_by_requester,
-            capsule_signature_by_requester_and_miner,
-            cfrag_bytes,
-            cfrag_signature_by_miner,
-            requester_umbral_public_key_bytes,
-            miner_umbral_public_key_bytes,
-            signed_miner_umbral_public_key,
-            some_data)
-
-    assert not challenge_contract.functions.challengedCFrags(data_hash).call()
-    tx = challenge_contract.functions.challengeCFrag(*args).transact()
-    testerchain.wait_for_receipt(tx)
-    # Hash of the data is saved and miner was slashed
-    assert challenge_contract.functions.challengedCFrags(data_hash).call()
-    assert 900 == escrow.functions.minerInfo(miner).call()
-
+    # TODO: Fix test for cfrag evaluation using bad data
```

Comment:
What's up here?

Reply:
It's not clear to me how to prepare a meaningful test for this right now, so I left it as future work. I don't want to have this PR open for so long.

Reply:
Can you open an issue for it and assign yourself?